SandWorm/MeshCompareComponent.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using Rhino.Geometry.Intersect;
+using Grasshopper.Kernel;
+using Rhino.Geometry;
+
+namespace SandWorm
+{
+    public class CompareMeshes
+    {
+        public CompareMeshes()
+        {
+        }
+
+        private double[] _meshElevationPoints;
+
+        public double[] MeshElevationPoints
+        {
+            get { return _meshElevationPoints; }
+            set { _meshElevationPoints = value; }
+        }
+    }
+    public class MeshCompareComponent : GH_Component
+    {
+        public List<GeometryBase> outputSurface;
+        private Curve inputRectangle;
+        private double scaleFactor = 1;
+        private double sensorElevation;
+        public int leftColumns = 0;
+        public int rightColumns = 0;
+        public int topRows = 0;
+        public int bottomRows = 0;
+        public SetupOptions options; // List of options coming from the SWSetup component
+        public CompareMeshes results; // Array of resulting elevation points 
+        public List<string> output;
+        public Mesh inputMesh;
+
+        public MeshCompareComponent()
+          : base("MeshCompare", "MeshCompare",
+              "Visualizes elevation differences between meshes.",
+              "Sandworm", "Sandbox")
+        {
+        }
+
+        /// <summary>
+        /// Registers all the input parameters for this component.
+        /// </summary>
+        protected override void RegisterInputParams(GH_Component.GH_InputParamManager pManager)
+        {
+            pManager.AddCurveParameter("Rectangle", "RC", "Rectangle", GH_ParamAccess.item);
```

Comment:
Current approach doesn't really care about the size or the aspect ratio of the rectangle. It only uses it to define the position and the rotation of the sampling plane. 

The fundamental design decision I had to make at the beginning was to either keep the reference mesh where it belongs to and manipulate the 'window' (our rectangle) to where we need it to be; or to keep the 'window' fixed and manipulate the mesh.

I went with the forme, having received your feedback, however, I think it might make more sense to go with the latter. What do you think?